# Part 2 Question 1

#Question 1: Between Donald Trump and Hillary Clinton, who tweeted their opponent's name most frequently during the 2016 US election?
The twitter data was downloaded from Kaggle. 
https://www.kaggle.com/benhamner/clinton-trump-tweets

In [ ]:
from google.colab import files 
  
uploaded = files.upload()

Saving tweets.csv to tweets.csv


In [ ]:
import pandas as pd 
import io

In [ ]:
data = pd.read_csv(io.BytesIO(uploaded['tweets.csv'])) 
data.head(6)

id  ...                                  extended_entities
0  780925634159796224  ...  {'media': [{'display_url': 'pic.twitter.com/Xr...
1  780916180899037184  ...  {'media': [{'display_url': 'pic.twitter.com/t0...
2  780911564857761793  ...                                                NaN
3  780907038650068994  ...  {'media': [{'display_url': 'pic.twitter.com/Q3...
4  780897419462602752  ...                                                NaN
5  780893126605037568  ...                                                NaN

[6 rows x 28 columns]

In [ ]:
trump_data = data.loc[data["handle"] == "realDonaldTrump"]
tweets_trump = trump_data[["handle","text"]]
tweets_trump.head(2)

handle                                               text
5  realDonaldTrump  Join me for a 3pm rally - tomorrow at the Mid-...
8  realDonaldTrump  Once again, we will have a government of, by a...

In [ ]:
tweets_trump = trump_data[["text"]]
tweets_trump.head(2)

text
5  Join me for a 3pm rally - tomorrow at the Mid-...
8  Once again, we will have a government of, by a...

In [ ]:
#How many tweets of Donald Trump mentioned Hillary Clinton's name?

mentioned_hillary = tweets_trump[tweets_trump['text'].str.match("[hH]illary*|[Cc]linton*")==True]
print(mentioned_hillary.count())
print(mentioned_hillary)

text    49
dtype: int64
                                                   text
12    Hillary Clinton's Campaign Continues To Make F...
50    Hillary's been failing for 30 years in not get...
94    Hillary Clinton failed all over the world. \n❌...
98    Hillary Clinton is the only candidate on stage...
100   Hillary's policies have made America less safe...
225   Hillary Clinton just lost every Republican she...
306   Hillary Clinton is taking the day off again, s...
331   Hillary Clinton's weakness while she was Secre...
668   Hillary Clinton just had her 47% moment. What ...
744   Hillary Clinton answered email questions diffe...
748   Hillary just gave a disastrous news conference...
935   Hillary Clinton doesn't have the strength or t...
948   Hillary Clinton didn't go to Louisiana, and no...
1090  Hillary Clinton only knows how to make a speec...
1096  Hillary Clinton is using race-baiting to try t...
1098  Hillary Clinton's short speech is pandering to...
1180  Hillary Clinton st

In [ ]:
hillary_data = data.loc[data["handle"] == "HillaryClinton"]
tweets_hillary = hillary_data[["handle","text"]]
tweets_hillary.head(2)

handle                                               text
0  HillaryClinton  The question in this election: Who can put the...
1  HillaryClinton  Last night, Donald Trump said not paying taxes...

In [ ]:
#How many tweets of Hillary Clinton mentioned Donald Trump's name?

mentioned_trump = tweets_hillary[tweets_hillary['text'].str.match("[dD]onald*|[tT]rump*") == True]
print(mentioned_trump.count())
print(mentioned_trump)

handle    263
text      263
dtype: int64
              handle                                               text
14    HillaryClinton  Donald Trump lied to the American people at le...
35    HillaryClinton  Trump sobre Alicia Machado en 1996: "Miss Pigg...
51    HillaryClinton  Trump on Alicia, 1996: "Miss Piggy."\n\nThis m...
73    HillaryClinton  Donald Trump called her "Miss Piggy" and "Miss...
90    HillaryClinton  Donald Trump's "cavalier attitude about nuclea...
...              ...                                                ...
4655  HillaryClinton  Donald Trump has a lot of ideas about foreign ...
4661  HillaryClinton  Donald Trump's anti-Muslim rhetoric goes again...
4768  HillaryClinton  Donald Trump is going to try to take Mitt Romn...
4785  HillaryClinton  Trump saying awful things about women isn't th...
4838  HillaryClinton  Donald Trump &amp; Ted Cruz are pushing a visi...

[263 rows x 2 columns]


In [ ]:
#How many tweets of Donald Trump mentioned Crooked Hillary? 
pd.options.display.max_rows = 5000
pd.options.display.max_colwidth = 50

mentioned_crooked_hillary = tweets_trump[tweets_trump['text'].str.match("[cC]rooked* [hH]illary*")==True]
print(mentioned_crooked_hillary.count())
print(mentioned_crooked_hillary)

text    67
dtype: int64
                                                   text
210   Crooked Hillary's bad judgement forced her to ...
312   Crooked Hillary has been fighting ISIS, or wha...
411   Crooked Hillary wants to take your 2nd Amendme...
879   Crooked Hillary's V.P. pick said this morning ...
1017  Crooked Hillary's brainpower is highly overrat...
1080  Crooked Hillary will NEVER be able to solve th...
1214  Crooked Hillary is flooding the airwaves with ...
1387  Crooked Hillary Clinton is being protected by ...
1602  Crooked Hillary said loudly, and for the world...
1724  Crooked Hillary Clinton is 100% owned by her d...
1792  Crooked Hillary Clinton is soft on crime, supp...
1835  Crooked Hillary said that I "couldn't handle t...
1839  Crooked Hillary Clinton mentioned me 22 times ...
1844  Crooked Hillary Clinton made up facts about me...
2109  Crooked Hillary Clinton wants to flood our cou...
2288  Crooked Hillary Clinton knew everything that h...
2302  Crooked Hillary Cl

**Hillary Clinton had more number of tweets mentioning Donald Trump's name (263 tweets) than the opposite (49 tweets). And, Crooked Hillary term was used by Donald Trump in 67 of his tweets.**

# Part 2 Question 2

# In Pirates of the Caribbean, how could you find the things said by a particular character (e.g., JACK SPARROW)?

In [ ]:
import nltk
nltk.download('webtext')
from nltk.corpus import webtext
webtext.fileids()

[nltk_data] Downloading package webtext to /root/nltk_data...
[nltk_data]   Unzipping corpora/webtext.zip.


['firefox.txt',
 'grail.txt',
 'overheard.txt',
 'pirates.txt',
 'singles.txt',
 'wine.txt']

In [ ]:
pirates = webtext.raw(fileids='pirates.txt')
print(pirates[0:3999]) #printing the first 4000 characters

PIRATES OF THE CARRIBEAN: DEAD MAN'S CHEST, by Ted Elliott & Terry Rossio
[view looking straight down at rolling swells, sound of wind and thunder, then a low heartbeat]
Scene: PORT ROYAL
[teacups on a table in the rain]
[sheet music on music stands in the rain]
[bouquet of white orchids, Elizabeth sitting in the rain holding the bouquet]
[men rowing, men on horseback, to the sound of thunder]
[EITC logo on flag blowing in the wind]
[many rowboats are entering the harbor]
[Elizabeth sitting alone, at a distance]
[marines running, kick a door in] 
[a mule is seen on the left in the barn where the marines enter]
[Liz looking over her shoulder]
[Elizabeth drops her bouquet]
[Will is in manacles, being escorted by red coats]
ELIZABETH SWANN: Will...!
[Elizabeth runs to Will]
ELIZABETH SWANN: Why is this happening? 
WILL TURNER: I don't know. You look beautiful.
ELIZABETH SWANN: I think it's bad luck for the groom to see the bride before the wedding.
[marines cross their long axes to bar Go

In [ ]:
#finding lines said by the character "Jack Sparrow"
import re
for line in re.findall('JACK SPARROW.+', pirates):
    print(line)

JACK SPARROW: Sorry, mate.
JACK SPARROW: Mind if we make a little side trip? I didn't think so.
JACK SPARROW: Complications arose, ensued, were overcome.
JACK SPARROW: Mm-hmm!
JACK SPARROW: Shiny?
JACK SPARROW: Is that how you're all feeling, then? Perhaps dear old Jack is not serving your best interests as captain?
JACK SPARROW: What did the bird say?
JACK SPARROW: Ohhh!
JACK SPARROW: It does me.
JACK SPARROW: No! Much more better. It is a *drawing* of a key. 
JACK SPARROW: Gentlemen, what do keys do?
JACK SPARROW: No! If we don't have the key, we can't open whatever it is we don't have that it unlocks. So what purpose would be served in finding whatever need be unlocked, which we don't have, without first having found the key what unlocks it?
JACK SPARROW: You're not making any sense at all. Any more questions?
JACK SPARROW: Hah! A heading. Set sail in a... mmm... a general... in *that* way - direction. 
JACK SPARROW: Come on, snap to and make sail, you know how this works. Come on, 

In [ ]:
#finding lines said by the character "ELIZABETH SWANN"
for line in re.findall('ELIZABETH SWANN.+', pirates):
    print(line)

ELIZABETH SWANN: Will...!
ELIZABETH SWANN: Why is this happening? 
ELIZABETH SWANN: I think it's bad luck for the groom to see the bride before the wedding.
ELIZABETH SWANN: On what charges?
ELIZABETH SWANN: *What are the charges?*
ELIZABETH SWANN: We are under the jurisdiction of the King's governor of Port Royal, and you will tell us what we are charged with. 
ELIZABETH SWANN: *Captain* Jack Sparrow.
ELIZABETH SWANN: Jack's Compass. What does Beckett want with that?
ELIZABETH SWANN: I have faith in you. Both of you. Where will you find him? 
ELIZABETH SWANN: Properly? 
ELIZABETH SWANN: If it weren't for these bars I'd have you already. 
ELIZABETH SWANN: I'll wait for you.
ELIZABETH SWANN: Why don't you tell me what's happening? 
ELIZABETH SWANN: No! Will has gone to find Jack!
ELIZABETH SWANN: He's a better man than you give him credit for. 
ELIZABETH SWANN: A fair trial for Will ends in a hanging.
ELIZABETH SWANN: Then what is?
ELIZABETH SWANN: I expect then that we can come to some